lesson3/bigForm1/polymer/src/js/app.js

```diff
@@ -46,37 +57,63 @@ var componentForm = {
 var fullAddress = {};
 
 function initAutocomplete() {
-  // Create the autocomplete object, restricting the search to geographical
-  // location types.
-  autocomplete = new google.maps.places.Autocomplete(
-      (document.getElementById('ship-address').$.input),
-      {types: ['geocode']});
-
-  // When the user selects an address from the dropdown, populate the address
-  // fields in the form.
-  autocomplete.addListener('place_changed', fillInAddress);
-}
+	// Create the autocomplete objects, restricting the search to geographical
+	// location types.
+	autocompleteInputs.forEach(function(element) {
+		element.element = document.querySelector(element.selector);
+
+		element.googleAutocomplete = new google.maps.places.Autocomplete(
+			(element.element.$.input),
+			{type: ['geocode']}
+		);
+		// When the user selects an address from the dropdown, populate the address
+		// fields in the form.
+		element.googleAutocomplete.addListener('place_changed', fillInAddress);
+		// When the user clears the input
+		element.element.addEventListener('blur', fillInAddress);
+	});
+};
 
+/*
+Determines how the zip field should be filled
+ */
 function fillInAddress() {
-  // Get the place details from the autocomplete object.
-  var place = autocomplete.getPlace();
-
-  // Get each component of the address from the place details
-  // and fill the corresponding field on the form.
-  for (var i = 0; i < place.address_components.length; i++) {
-    var addressType = place.address_components[i].types[0];
-
-    if (componentForm[addressType]) {
-      var val = place.address_components[i][componentForm[addressType]];
-	    fullAddress[addressType] = val;
-	    if (addressType === "postal_code") {
-	    	var zipInput = document.querySelector('gold-zip-input')
-	    	zipInput.value = val;
-	    	zipInput.oninput();
-	    }
-    }
-  }
-}
+
+	var priority = 0;
+	var target, googleAutocomplete;
+	var zipInput = document.querySelector('gold-zip-input');
+
+	// Clear value in case all inputs are emptied
+	zipInput.value = '';
```

Comment:
FYI, I'll be removing this line from the release. This forces the zip input to refresh every time either addresses change even if the resulting input stays the same. It's also not a big deal if the zip is left behind when both inputs are cleared.